instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/HttpServerTracer.java

```diff
@@ -239,9 +235,31 @@ static String extractForwardedFor(String forwarded) {
     if (start >= forwarded.length() - 1) { // the value after for= must not be empty
       return null;
     }
+    return extractIpAddress(forwarded, start);
+  }
+
+  // VisibleForTesting
+  static String extractForwardedFor(String forwarded) {
+    return extractIpAddress(forwarded, 0);
+  }
+
+  private static String extractIpAddress(String forwarded, int start) {
+    if (forwarded.length() == start) {
+      return null;
+    }
+    if (forwarded.charAt(start) == '[') {
```

Comment:
Looking at https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded and https://www.rfc-editor.org/rfc/rfc7239 it seems to me that ipv6 addresses need to be quoted, for example `for="[2001:db8:cafe::17]"`. Similarly when ipv4 address includes a port it should be quoted `"192.0.2.43:47011"`

Reply:
good catch, fixed, and yet more tests added...